src/warnet/graph.py

```diff
@@ -17,3 +24,159 @@ def import_json(infile: Path, outfile: Path, cb: str, ln_image: str):
     Returns XML file as string with or without --outfile option.
     """
     raise Exception("Not Implemented")
+
+
+def custom_graph(num_nodes: int, num_connections: int, version: str, datadir: Path):
+    datadir.mkdir(parents=False, exist_ok=False)
+    # Generate network.yaml
+    nodes = []
+    connections = set()
+
+    for i in range(num_nodes):
+        node = {"name": f"tank-{i:04d}", "connect": [], "image": {"tag": version}}
+
+        # Add round-robin connection
+        next_node = (i + 1) % num_nodes
+        node["connect"].append(f"tank-{next_node:04d}")
+        connections.add((i, next_node))
+
+        # Add random connections
+        available_nodes = list(range(num_nodes))
+        available_nodes.remove(i)
+        if next_node in available_nodes:
+            available_nodes.remove(next_node)
+
+        for _ in range(min(num_connections - 1, len(available_nodes))):
+            # trunk-ignore(bandit/B311)
+            random_node = random.choice(available_nodes)
+            # Avoid circular loops of A -> B -> A
+            if (random_node, i) not in connections:
+                node["connect"].append(f"tank-{random_node:04d}")
+                connections.add((i, random_node))
+                available_nodes.remove(random_node)
+
+        nodes.append(node)
+
+    network_yaml_data = {"nodes": nodes}
+
+    with open(os.path.join(datadir, "network.yaml"), "w") as f:
+        yaml.dump(network_yaml_data, f, default_flow_style=False)
+
+    # Generate defaults.yaml
+    default_yaml_path = files("resources.networks").joinpath("6_node_bitcoin/node-defaults.yaml")
```

Comment:
note to self: we should move all the hardcoded values into something like config.py or constants.py , so these can be something like `default_yaml_path = config.DEFAULT_YAML_PATH` etc